lib/IntermediateRepresentation/basicblock.cc

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "program.h"
+#include "stmt.h"
+
+namespace Fortran::IntermediateRepresentation {
```

Comment:
I don't like "IntermediateRepresentation" as a namespace name. We don't have a convention for these and I couldn't find one in the LLVM Coding Standards, but Google mandates lower case. And it will be painful to use outside of this namespace.

I think "ir" is fine.

Reply:
I wanted to call this FIR, but that was shot down as a violation of https://github.com/flang-compiler/f18/blob/master/documentation/C%2B%2Bstyle.md and its prohibition of using abbreviations. IR clearly fails to get over that bar too.  Furthermore, I definitely do not want to call this representation "IR" as there already is a more well-known IR (LLVM's), and it is hard to see anything but disadvantages to naming two distinct and separate aspects of the compiler the same name.  Finally, a workaround for long namespaces is available, as one can always do `using FIR = Fortran::IntermediateRepresentation`.

Reply:
The style guide discourages abbreviations but doesn't prohibit them. In this case, I think "fir" is good and it's okay that everyone who works on the compiler has to learn what that means. I think namespace names should be all lower case.